dist/images/ovnkube.sh

```diff
@@ -1407,47 +1406,420 @@ ovnkube-controller() {
-    --logfile-maxsize=${ovnkube_logfile_maxsize} \
-    --logfile-maxbackups=${ovnkube_logfile_maxbackups} \
-    --logfile-maxage=${ovnkube_logfile_maxage} \
-    ${hybrid_overlay_flags} \
+  /usr/bin/ovnkube --init-ovnkube-controller ${K8S_NODE} \
+    ${anp_enabled_flag} \
     ${disable_snat_multiple_gws_flag} \
+    ${egressfirewall_enabled_flag} \
+    ${egressip_enabled_flag} \
+    ${egressip_healthcheck_port_flag} \
+    ${egressqos_enabled_flag} \
+    ${egressservice_enabled_flag} \
     ${empty_lb_events_flag} \
+    ${hybrid_overlay_flags} \
+    ${libovsdb_client_logfile_flag} \
+    ${multicast_enabled_flag} \
+    ${multi_network_enabled_flag} \
+    ${ovn_acl_logging_rate_limit_flag} \
+    ${ovn_dbs} \
+    ${ovnkube_config_duration_enable_flag} \
+    ${ovnkube_enable_interconnect_flag} \
+    ${ovnkube_enable_multi_external_gateway_flag} \
+    ${ovnkube_metrics_tls_opts} \
+    ${ovn_master_ssl_opts} \
     ${ovn_v4_join_subnet_opt} \
-    ${ovn_v6_join_subnet_opt} \
     ${ovn_v4_masquerade_subnet_opt} \
+    ${ovn_v6_join_subnet_opt} \
     ${ovn_v6_masquerade_subnet_opt} \
-    --pidfile ${OVN_RUNDIR}/ovnkube-controller.pid \
+    --cluster-subnets ${net_cidr} --k8s-service-cidr=${svc_cidr} \
+    --gateway-mode=${ovn_gateway_mode} \
+    --host-network-namespace ${ovn_host_network_namespace} \
+    --logfile-maxage=${ovnkube_logfile_maxage} \
+    --logfile-maxbackups=${ovnkube_logfile_maxbackups} \
+    --logfile-maxsize=${ovnkube_logfile_maxsize} \
     --logfile /var/log/ovn-kubernetes/ovnkube-controller.log \
-    ${libovsdb_client_logfile_flag} \
-    ${ovn_master_ssl_opts} \
-    ${ovnkube_metrics_tls_opts} \
-    ${multicast_enabled_flag} \
-    ${ovn_acl_logging_rate_limit_flag} \
+    --loglevel=${ovnkube_loglevel} \
+    --metrics-bind-address ${ovnkube_master_metrics_bind_address} \
+    --pidfile ${OVN_RUNDIR}/ovnkube-controller.pid \
+    --zone ${ovn_zone} &
+
+  echo "=============== ovnkube-controller ========== running"
+  wait_for_event attempts=3 process_ready ovnkube-controller
+
+  process_healthy ovnkube-controller
+  exit 9
+}
+
+ovnkube-controller-with-node() {
+  trap 'kill $(jobs -p) ; rm -f /etc/cni/net.d/10-ovn-kubernetes.conf ; exit 0' TERM
+  check_ovn_daemonset_version "3"
+  rm -f ${OVN_RUNDIR}/ovnkube-controller-with-node.pid
+
+  if [[ ${ovnkube_node_mode} != "dpu-host" ]]; then
+    echo "=============== ovnkube-controller-with-node - (wait for ovs)"
+    wait_for_event ovs_ready
+  fi
+
+  echo "=============== ovnkube-controller-with-node (wait for ready_to_start_node) =========="
+  wait_for_event ready_to_start_node
+  echo "ovn_nbdb ${ovn_nbdb}   ovn_sbdb ${ovn_sbdb}  ovn_nbdb_conn ${ovn_nbdb_conn}"
+
+  # wait for northd to start
+  wait_for_event process_ready ovn-northd
+
+  # wait for ovs-servers to start since ovn-master sets some fields in OVS DB
+  echo "=============== ovnkube-controller-with-node - (wait for ovs)"
+  wait_for_event ovs_ready
```

Comment:
Is this a duplicate of line 1459 above?

Reply:
I think this is a bad merge of ovnkube-node and ovnkube-controller start functions